src/react/src/components/Contribute/ProductionLocationInfo.jsx

```diff
@@ -112,418 +262,618 @@ const ProductionLocationInfo = ({
-                        className={classes.submitButtonStyles}
-                    >
-                        Submit
-                    </Button>
-                </div>
-            </Paper>
-        </div>
+                                    <Typography
+                                        component="h2"
+                                        className={classes.titleStyles}
+                                    >
+                                        Parent Company
+                                    </Typography>
+                                    <Typography
+                                        component="h4"
+                                        className={classes.subTitleStyles}
+                                    >
+                                        Enter the company that holds majority
+                                        ownership for this production.
+                                    </Typography>
+                                    <StyledSelect
+                                        creatable
+                                        name="Parent company"
+                                        value={parentCompany}
+                                        onChange={setParentCompany}
+                                        placeholder="Enter the parent company"
+                                        aria-label="Parent company"
+                                        styles={selectStyles}
+                                        className={classes.selectStyles}
+                                        components={customSelectComponents}
+                                    />
+                                </div>
+                            </>
+                        )}
+                    </div>
+                    <div className={classes.buttonsContainerStyles}>
+                        <Button
+                            variant="outlined"
+                            onClick={handleGoBack}
+                            className={classes.goBackButtonStyles}
+                        >
+                            Go Back
+                        </Button>
+                        <Button
+                            color="secondary"
+                            variant="contained"
+                            onClick={() => {
+                                const data = {
+                                    name: inputName,
+                                    address: inputAddress,
+                                    country: inputCountry,
+                                    sector,
+                                    productType,
+                                    locationType,
+                                    processingType,
+                                    numberOfWorkers,
+                                    parentCompany,
+                                };
+                                handleProductionLocation(data, osID);
```

Comment:
Can you simplify the onClick by moving the data object outside?

Reply:
Done.